azavea/echo-locator
# Add neighborhood models and import script

## Overview

Adds models to store neighborhood information and an import script to populate them from `neighborhoods.json` and `neighborhood_bounds.json`. Photos are not handled yet (#494). Also switches maps on the backend to use OSM rather than satellite basemaps.

### Checklist


### Demo

<img width="862" alt="Screen Shot 2022-05-31 at 08 27 36" src="https://user-images.githubusercontent.com/447977/171173326-a9beca6b-1702-430d-84a7-62e4964dd487.png">
<img width="904" alt="Screen Shot 2022-05-31 at 08 27 26" src="https://user-images.githubusercontent.com/447977/171173328-9f8f5f00-39d9-41b8-9d84-d9c57b390efc.png">
<img width="844" alt="Screen Shot 2022-05-31 at 08 27 09" src="https://user-images.githubusercontent.com/447977/171173332-be8eaa5f-b6c5-4950-914a-0c1663efa7fa.png">


### Notes

- I chose not to do much error-handling in the import script because while the neighborhood data is ultimately human-generated, it passes through a number of scripts in order to be turned into the JSON files that the import script consumes, so those should catch most errors and leave the data in a pretty good state by the time it reaches this import script.
- The script is designed to update existing zipcodes if they exist because BHA has said that in the future, they would like to carry out data updates once per year. This also makes error correction easier because the script can be re-run multiple times and will update existing neighborhoods and/or create new ones, as needed.

## Testing Instructions

 * Run `./scripts/update` to run migrations, and then `./scripts/server`.
 * Navigate to `localhost:8085/admin/` and log in.
 * Edit a user profile and confirm that the basemap is OSM streets rather than a satellite view.
 * Run `AWS_PROFILE=echo-locator ./scripts/manage import_neighborhood_bounds s3://echolocator-production-data-us-east-1/neighborhoods.json s3://echolocator-production-data-us-east-1/neighborhood_bounds.json` (the files have already been uploaded to S3). Confirm that it runs without errors and reports that all Neighborhoods were newly created.
 * Run the script again and confirm that it reports that neighborhoods were updated.
 * Spot-check some neighborhoods using the Django Admin (and maybe geojson.io) and confirm that the values, polygon, and centroid match the values from the GeoJSON.


Resolves #492
